apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -917,48 +917,49 @@ func (m *ModelManagerAPI) ModelInfo(ctx context.Context, args params.Entities) (
 			valid := !cred.Invalid
 			modelInfo.CloudCredentialValidity = &valid
 		}
-		if canWrite {
-			st, release, err := m.state.GetBackend(tag.Id())
-			if errors.Is(err, errors.NotFound) {
-				return params.ModelInfo{}, errors.Trace(apiservererrors.ErrPerm)
-			} else if err != nil {
-				return params.ModelInfo{}, errors.Trace(err)
-			}
-			defer release()
+		if !canWrite {
+			return modelInfo, nil
+		}
 
-			modelUUID := coremodel.UUID(tag.Id())
-			modelDomainServices, err := m.domainServicesGetter.DomainServicesForModel(ctx, modelUUID)
-			if err != nil {
-				return params.ModelInfo{}, errors.Trace(err)
-			}
-			// TODO: remove mongo state from the commonmodel.ModelMachineInfo.
-			if modelInfo.Machines, err = commonmodel.ModelMachineInfo(ctx, st, modelDomainServices.Machine()); err != nil {
-				return params.ModelInfo{}, err
-			}
+		st, release, err := m.state.GetBackend(tag.Id())
+		if errors.Is(err, errors.NotFound) {
+			return params.ModelInfo{}, errors.Trace(apiservererrors.ErrPerm)
+		} else if err != nil {
+			return params.ModelInfo{}, errors.Trace(err)
+		}
+		defer release()
 
-			backends, err := m.secretBackendService.BackendSummaryInfoForModel(ctx, modelUUID)
-			if err != nil {
-				return params.ModelInfo{}, errors.Trace(err)
-			}
-			for _, backend := range backends {
-				name := backend.Name
-				if name == kubernetes.BackendName {
-					name = kubernetes.BuiltInName(modelInfo.Name)
-				}
-				modelInfo.SecretBackends = append(modelInfo.SecretBackends, params.SecretBackendResult{
-					// Don't expose the id.
-					NumSecrets: backend.NumSecrets,
-					Status:     backend.Status,
-					Message:    backend.Message,
-					Result: params.SecretBackend{
-						Name:                name,
-						BackendType:         backend.BackendType,
-						TokenRotateInterval: backend.TokenRotateInterval,
-						Config:              backend.Config,
-					},
-				})
-			}
+		modelUUID := coremodel.UUID(tag.Id())
+		modelDomainServices, err := m.domainServicesGetter.DomainServicesForModel(ctx, modelUUID)
+		if err != nil {
+			return params.ModelInfo{}, errors.Trace(err)
+		}
+		// TODO: remove mongo state from the commonmodel.ModelMachineInfo.
+		if modelInfo.Machines, err = commonmodel.ModelMachineInfo(ctx, st, modelDomainServices.Machine()); err != nil {
+			return params.ModelInfo{}, err
+		}
 
+		backends, err := m.secretBackendService.BackendSummaryInfoForModel(ctx, modelUUID)
```

Comment:
nit: maybe infos would be better since we seem to use backend to get different state entities previously.

Reply:
I may change to secretBackends.